cpp/src/ray/util/process_helper.cc

```diff
@@ -100,13 +83,14 @@ void ProcessHelper::RayStart(CoreWorkerOptions::TaskExecutionCallback callback)
     }
   }
 
+  std::unique_ptr<ray::gcs::GlobalStateAccessor> global_state_accessor = nullptr;
   if (ConfigInternal::Instance().worker_type == ray::WorkerType::DRIVER) {
-    ray::gcs::GlobalStateAccessor global_state_accessor(
-        redis_address, ConfigInternal::Instance().redis_password);
-    RAY_CHECK(global_state_accessor.Connect()) << "Failed to connect to GCS.";
+    global_state_accessor.reset(new ray::gcs::GlobalStateAccessor(
```

Comment:
Use an object is better than a pointer here.

Reply:
We don't always init this object so a pointer is convenient for check if it's initialized.